sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py

```diff
@@ -143,50 +142,98 @@ def from_connection_string(cls, conn_str: str, **kwargs) -> "ServiceBusManagemen
+        :keyword enable_batched_operations: Value that indicates whether server-side batched operations
+         are enabled.
+        :type enable_batched_operations: bool
+        :keyword enable_express: A value that indicates whether Express Entities are enabled. An express
+         queue holds a message in memory temporarily before writing it to persistent storage.
+        :type enable_express: bool
+        :keyword enable_partitioning: A value that indicates whether the queue is to be partitioned
+         across multiple message brokers.
+        :type enable_partitioning: bool
+        :keyword is_anonymous_accessible: A value indicating if the resource can be accessed without
+         authorization.
+        :type is_anonymous_accessible: bool
+        :keyword lock_duration: ISO 8601 timespan duration of a peek-lock; that is, the amount of time
+         that the message is locked for other receivers. The maximum value for LockDuration is 5
+         minutes; the default value is 1 minute.
+        :type lock_duration: ~datetime.timedelta
+        :keyword max_delivery_count: The maximum delivery count. A message is automatically deadlettered
+         after this number of deliveries. Default value is 10.
+        :type max_delivery_count: int
+        :keyword max_size_in_megabytes: The maximum size of the queue in megabytes, which is the size of
+         memory allocated for the queue.
+        :type max_size_in_megabytes: int
+        :keyword requires_duplicate_detection: A value indicating if this queue requires duplicate
+         detection.
+        :type requires_duplicate_detection: bool
+        :keyword requires_session: A value that indicates whether the queue supports the concept of
+         sessions.
+        :type requires_session: bool
+        :keyword forward_to: The name of the recipient entity to which all the messages sent to the queue
+         are forwarded to.
+        :type forward_to: str
+        :keyword user_metadata: Custom metdata that user can associate with the description. Max length
+         is 1024 chars.
+        :type user_metadata: str
+        :keyword support_ordering: A value that indicates whether the queue supports ordering.
+        :type support_ordering: bool
+        :keyword forward_dead_lettered_messages_to: The name of the recipient entity to which all the
+         dead-lettered messages of this subscription are forwarded to.
+        :type forward_dead_lettered_messages_to: str
+
+        :rtype: ~azure.servicebus.management.QueueProperties
         """
-        try:
-            queue_name = queue.name  # type: ignore
-            to_create = queue._to_internal_entity()  # type: ignore
-        except AttributeError:
-            queue_name = queue  # type: ignore
-            to_create = InternalQueueDescription()  # Use an empty queue description.
-
+        queue = QueueProperties(name, **kwargs)
+        for key in queue.keys():
+            kwargs.pop(key, None)
```

Comment:
should this be part of the logic embedded in QueueProperties constructor? I recall uAMQP/EH doing something similar in the constructor.

Reply:
As per our discussion and test, use pop( ) in QueueProperties constructor doesn't pop the keywords in the original kwargs.